tools/clang/test/HLSLFileCheck/hlsl/intrinsics/basic/fmod_const_eval.hlsl

```diff
@@ -0,0 +1,55 @@
+// RUN: %dxc -T vs_6_0 %s -E main | %FileCheck %s
+// RUN: %dxc -T vs_6_0 %s -E main -DNO_FOLD | %FileCheck %s -check-prefixes=NO_FOLD
+
+// Ensure fmod is constant evaluated during codegen, or dxil const eval
+// TODO: handle fp specials properly!
```

Comment:
The way we've been dealing with such is with XFAILS and bugs filed to account for them. I think that would require this to be a lit test, but I don't see any reason why it can't be.